### ISSUE TITLE
Loading dstu2 timing objects ends in an invalid state

### ISSUE THREAD

**Issue by ItayGoren** [hapifhir/org.hl7.fhir.core#548]
Looking at "timing" [specification in dstu2 version](https://www.hl7.org/fhir/DSTU2/datatypes.html#timing) we can see that `bounds` can be `Duration`, `Range` and `Period`. In the code, the class is defined to these classes.
The case where `duration` is used is saved in the variable `boundsQuantity` and not `boundsDuration`. It was [solved in r4](https://www.hl7.org/fhir/datatypes.html#Timing) by changing the field to be named `boundsDuration`.
Yet, when loading the object using `org.hl7.fhir.dstu2.formats.JsonParser().parse(resource_string)` we call `JsonParser.parseTiming` ([line 805](https://raw.githubusercontent.com/hapifhir/org.hl7.fhir.core/master/org.hl7.fhir.dstu2/src/main/java/org/hl7/fhir/dstu2/formats/JsonParser.java)) which calls `parseTimingTimingRepeatComponent` which parse `bounds` using `parseType("bounds", json)`.
The problem is that `parseType` check the suffix of the attribute, and in our case it is `Quantity`. Hence, the object is loaded as `Quantity` even though it is specified with `type = {Duration.class, Range.class, Period.class}`.
Hence, the state is invalid.

My problem with this case is when we convert `Timing` from `dstu2` to `r4`.
In this case, we convert `dstu2.Timing` to `r4.Timing` which set `bounds` using 
```python
  public static org.hl7.fhir.dstu2.model.Timing.TimingRepeatComponent convertTimingRepeatComponent(org.hl7.fhir.r4.model.Timing.TimingRepeatComponent src) throws FHIRException {
    if (src == null || src.isEmpty()) return null;
    org.hl7.fhir.dstu2.model.Timing.TimingRepeatComponent tgt = new org.hl7.fhir.dstu2.model.Timing.TimingRepeatComponent();
    copyElement(src, tgt);
    if (src.hasBounds()) tgt.setBounds(convertType(src.getBounds()));
```
Calling `convertType(src.getBounds())` with `dstu2.Quantity` would result in `r4.Quantity`. Hence, we call `tgt.setBounds` with `r4.Quantity`.
The following is `setBounds`:
```python
    public TimingRepeatComponent setBounds(Type value) { 
      if (value != null && !(value instanceof Duration || value instanceof Range || value instanceof Period))
        throw new Error("Not the right type for Timing.repeat.bounds[x]: "+value.fhirType());
      this.bounds = value;
      return this;
    }
```
As you see, it raise an `Error` in this case.

To summarise, when converting `Timing` resource from `dstu2` to `r4`, if it has `boundsQuantity`, an error would be raised.

A proposal PR was opened: https://github.com/hapifhir/org.hl7.fhir.core/pull/549

**Comment by markiantorno** [comment]
I requested changes on the PR

**Comment by markiantorno** [comment]
I'm closing this for now, if you want to re-open and continue work please do.
